greenmail-core/src/main/java/com/icegreen/greenmail/Managers.java

```diff
@@ -4,21 +4,104 @@
  */
 package com.icegreen.greenmail;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Type;
+
+import com.icegreen.greenmail.configuration.GreenMailConfiguration;
+import com.icegreen.greenmail.filestore.UncheckedFileStoreException;
 import com.icegreen.greenmail.imap.ImapHostManager;
 import com.icegreen.greenmail.imap.ImapHostManagerImpl;
 import com.icegreen.greenmail.smtp.SmtpManager;
-import com.icegreen.greenmail.store.InMemoryStore;
+import com.icegreen.greenmail.store.Store;
 import com.icegreen.greenmail.user.UserManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * @author Wael Chatila
  * @version $Id: $
  * @since Jan 27, 2006
  */
 public class Managers {
-    private ImapHostManager imapHostManager = new ImapHostManagerImpl(new InMemoryStore());
-    private UserManager userManager = new UserManager(imapHostManager);
-    private SmtpManager smtpManager = new SmtpManager(imapHostManager, userManager);
+    final Logger log = LoggerFactory.getLogger(Managers.class);
+
+    private ImapHostManager imapHostManager = null;
+    private UserManager userManager = null;
+    private SmtpManager smtpManager = null;
+    private Store storeToUse = null;
+
+    /**
+     * Public constructor wihtout Startup Configuration
+     *
+     * @deprecated Please use the constructor with a valid Startup Configuration
+     */
+    public Managers() {
+        this(new GreenMailConfiguration());
+    }
+
+    /**
+     * Public constructor with startupConfiguration
+     *
+     * @param startupConfig - the startup configuration
+     */
+    public Managers(GreenMailConfiguration startupConfig) {
+        log.info("Starting managers with the following startup configuration:");
+        startupConfig.logConfiguration();
+
+        // Depending on the startup confifguration, construct the correct Store class using Reflection:
+        this.constructStore(startupConfig);
+
+        this.imapHostManager = new ImapHostManagerImpl(this.storeToUse);
+        this.userManager = new UserManager(imapHostManager);
+        this.smtpManager = new SmtpManager(imapHostManager, userManager, startupConfig);
+    }
+
+    /**
+     * Creates the store class (InMemory or File-based), depending on the Startup Properties.
+     *
+     * @param startupConfig
+     */
+    private void constructStore(GreenMailConfiguration startupConfig) {

```

Comment:
I don't think creation of any type of "store" should be in this class. The primary purpose of this class is to bootstrap different types of managers for SMTP,IMAP and Users.